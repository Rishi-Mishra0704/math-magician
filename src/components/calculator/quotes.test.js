import React from 'react';
import {
  render, waitFor, act, screen,
} from '@testing-library/react';
import Quote from './quotes';
import '@testing-library/jest-dom/extend-expect';

describe('Quote', () => {
  beforeEach(() => {
    jest.spyOn(global, 'fetch').mockResolvedValue({
      json: jest.fn().mockResolvedValue([{ quote: 'test quote' }]),
    });
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should render loading text', () => {
    act(() => {
      render(<Quote />);
    });
    expect(screen.getByText(/Loading/i)).toBeInTheDocument();
  });

  it('should render error text when fetching quote fails', async () => {
    jest.spyOn(global, 'fetch').mockRejectedValue(new Error('Failed to fetch quote.'));
    act(() => {
      render(<Quote />);
    });
    await waitFor(() => expect(screen.getByText(/Failed to fetch quote/i)).toBeInTheDocument());
  });

  it('should render quote text when fetching quote succeeds', async () => {
    act(() => {
      render(<Quote />);
    });
    await waitFor(() => expect(screen.getByText(/test quote/i)).toBeInTheDocument());
  });
});
