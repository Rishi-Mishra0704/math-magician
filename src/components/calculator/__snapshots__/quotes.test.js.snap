// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote should render error text when fetching quote fails 1`] = `<DocumentFragment />`;

exports[`Quote should render loading text 1`] = `<DocumentFragment />`;

exports[`Quote should render quote text when fetching quote succeeds 1`] = `<DocumentFragment />`;
